pg_mon.c

```diff
@@ -635,6 +645,12 @@ static mon_rec * create_or_get_entry(mon_rec temp_entry, int64 queryId)
         {
             *entry = temp_entry;
             SpinLockInit(&entry->mutex);
+
+            /* Since this is a new query,  log the query text */
+            if (CONFIG_LOG_QUERY)
+            {
+                elog(LOG, "%s", queryDesc->sourceText);
```

Comment:
Probably it's worth it to mark the query somehow more explicitly as "logged from pg_mon for the purpose of ...". Otherwise I can imagine it would be pretty confusing when one does log digging (e.g. multiple appearances of the same query in different contexts). Also, any particular reason to not use `ereport`?

Another minor concern is the name of configuration option. Outside of GUC definition there is no context what kinds of queries are logged and why. Maybe at least `CONFIG_LOG_NEW_QUERIES`?